ext/sqlite3/sqlite3.c

```diff
@@ -1796,6 +1804,7 @@ PHP_METHOD(SQLite3Stmt, execute)
 			result->stmt_obj = stmt_obj;
 			result->column_names = NULL;
 			result->column_count = -1;
+			stmt_obj->pending_step_return_code = return_code;
 			ZVAL_OBJ_COPY(&result->stmt_obj_zval, Z_OBJ_P(object));
```

Comment:
Note this OBJ_COPY stmt_obj_zval for my next comment... I'm a bit out of my depth on the refcounting here and I'm not sure I'm totally clear on what's happening. We're creating a new SQLite3Result object here, and in it we're storing a zval that is an object pointer to our statement object. This seems fine so far... we want to increase the refcount on the statement so that it isn't cleaned up as the result object will need data from it... right?